### PR TITLE
Mman2 modification

### PR DESCRIPTION
**Pull request by jimartin361**
Attempt to fix emulated mmap behavior when it is passed an address in the first argument.

### CODE REVIEW

File: qiling/os/posix/syscall/mman.py [hunk 1]
```diff
@@ -210,8 +210,7 @@ def ql_syscall_mmap2(ql, mmap2_addr, mmap2_length, mmap2_prot, mmap2_flags, mmap
 
     mmap_base = mmap2_addr
     need_mmap = True
-
-    if mmap2_addr != 0 and mmap2_addr < ql.loader.mmap_address:
+    if mmap2_addr !=0 and ql.mem.is_mapped(mmap2_addr,mmap2_length):
```
Comment (aquynh) [inline qiling/os/posix/syscall/mman.py:213]: please put a space after `,`

**Comment by xwings** [comment]
changed base to dev

**Comment by xwings** [comment]
there is a conflict. will you be able to fix it ?

**Comment by xwings** [comment]
I will merge and fix the conflict

**Pull request merged**